src/fam.rs

```diff
@@ -430,10 +430,31 @@ impl<T: Default + FamStruct> PartialEq for FamStructWrapper<T> {
 
 impl<T: Default + FamStruct> Clone for FamStructWrapper<T> {
     fn clone(&self) -> Self {
```

Comment:
I think we can improve the previous implementation by also copying the header. Something like:

```
// It is safe to unwrap() for now since `self` is a valid `FamStructWrapper`, so the number
// of entries can't be > T::max_len().
let mut clone = FamStructWrapper::from_entries(self.as_slice()).unwrap();
clone.mem_allocator[0] = unsafe { std::ptr::read(self.as_fam_struct_ref()) };

clone
```

Would there be any concern related to this ?

Reply:
I want to avoid the `unwrap()` because if in the future `from_entries()` will return some other error as well, there's a high chance we're going to miss fixing this `unwrap()` and stuff will become risky. Also with your change, we are actually doing more things than we need.
I would also like to avoid the duplicate code that I added here, but didn't find a decent way to do that and not make things awkward.

Reply:
With your previous proposal, the things that were not needed would be:
- checking that the number of entries > max_len()
- pushing T::default() and at the end updating with the Fam value instead of directly pushing this value
- explicitly setting the `len` field of the Fam: `mem_allocator[0].set_len(num_elements);` which if I understand correctly, shouldn't be needed anymore if `len` is read from `self`